internal/scorecard/storage.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scorecard
+
+import (
+	"archive/tar"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/kubectl/pkg/scheme"
+)
+
+const (
+	StorageSidecarContainer = "scorecard-gather"
+)
+
+func (r PodTestRunner) execInPod(podName, mountPath, containerName string) (io.Reader, error) {
+	cmd := []string{
+		"tar",
+		"cf",
+		"-",
+		mountPath,
+	}
+
+	reader, outStream := io.Pipe()
+	const tty = false
+	req := r.Client.CoreV1().RESTClient().Post().
+		Resource("pods").
+		Name(podName).
+		Namespace(r.Namespace).SubResource("exec").Param("container", containerName)
+	req.VersionedParams(
+		&v1.PodExecOptions{
+			Command: cmd,
+			Stdin:   false,
+			Stdout:  true,
+			Stderr:  true,
+			TTY:     tty,
+		},
+		scheme.ParameterCodec,
+	)
+
+	var stderr bytes.Buffer
+	exec, err := remotecommand.NewSPDYExecutor(r.RESTConfig, "POST", req.URL())
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
+		defer outStream.Close()
+		err = exec.Stream(remotecommand.StreamOptions{
+			Stdin:  nil,
+			Stdout: outStream,
+			Stderr: &stderr,
+		})
+		if err != nil {
+			log.Error(err)
+		}
+	}()
+	return reader, err
+}
+
+func getStoragePrefix(file string) string {
+	return strings.TrimLeft(file, "/")
+}
+
+func untarAll(reader io.Reader, destDir, prefix string) error {
+	tarReader := tar.NewReader(reader)
+	for {
+		header, err := tarReader.Next()
+		if err != nil {
+			if err != io.EOF {
+				return err
+			}
+			break
+		}
+
+		if !strings.HasPrefix(header.Name, prefix) {
+			return fmt.Errorf("tar contents corrupted")
+		}
+
+		mode := header.FileInfo().Mode()
+		destFileName := filepath.Join(destDir, header.Name[len(prefix):])
+
+		baseName := filepath.Dir(destFileName)
+		if err := os.MkdirAll(baseName, 0755); err != nil {
+			return err
+		}
+		if header.FileInfo().IsDir() {
+			if err := os.MkdirAll(destFileName, 0755); err != nil {
+				return err
+			}
+			continue
+		}
+		evaledPath, err := filepath.EvalSymlinks(baseName)
+		if err != nil {
+			return err
+		}
+
+		if mode&os.ModeSymlink != 0 {
+			linkname := header.Linkname
+
+			if !filepath.IsAbs(linkname) {
+				_ = filepath.Join(evaledPath, linkname)
```

Comment:
What's this for?

Reply:
looks like nothing to me, I can't imagine this is doing anything